wallet-new/src/Cardano/Wallet/Kernel.hs

```diff
@@ -119,8 +122,14 @@ handlesOpen mode =
             metadb <- openMetaDB ":memory:"
             migrateMetaDB metadb
             return $ Handles db metadb
-        UseFilePath (DatabasePaths acidDb sqliteDb) -> do
+        UseFilePath (DatabaseOptions acidDb sqliteDb rebuildDB) -> do
+            let deleteMaybe fp = do
+                    when rebuildDB $ do
+                        itsHere <- doesDirectoryExist fp
+                        when itsHere $ removeDirectoryRecursive fp
```

Comment:
To play the devil's advocate here, my gut feeling suggests that we might be able to separate the concerns of the server initialisation (where we deal with all those crazy CLI options) with the pure, nice internals we have at the moment. Making this `rebuildDB` option leak into the `DatabasePaths` feels a bit artificial to me in that sense.

I think a better solution might be to do this rebuilding _before_ entering `bracketPassiveWallet`. What do you think? This way, we can do everything inside `Cardano.Wallet.Server.Action` rather than here.

Reply:
Yeah, I can see both sides of the coin. I don't think that's a massive deal breaker, and something we can modify/improve at a later stage.